### PR TITLE
[MRG] fix `CITATION.cff` YAML and add a test for parseability and content.

### LINKED ISSUE #2083: CITATION.cff is not valid YAML; this blocks zenodo DOI generation on releases.
Continuation of #1806 - I contacted zenodo support to ask why DOIs weren't being generated for new sourmash releases, and they said:

>Dear Titus,
The releases are failing because the `CITATION.cff` file is not a
valid yaml.
The error I can see in the system is:
```
File "/opt/zenodo/lib/python2.7/site-packages/yaml/scanner.py",
line 579, in fetch_value
self.get_mark())
ScannerError: mapping values are not allowed here
in "", line 3, column 16:
title: sourmash: a library for MinHash sketchin ...
```
>I have re-triggered manually that release, to skip errors, so you
will see it published in Zenodo. However, subsequent releases
will keep failing until it is fixed.Unfortunately, at the moment
we have no method to expose this error to the users. We are sorry   
for the inconvenience.

>Please let me know if there's anything else I can help you with.

>Best regards,
Pablo
Zenodo Support
https://zenodo.org ( https://zenodo.org )      

---

and of course I can reproduce the error with:

```
>>> import yaml
>>> with open('CITATION.cff', 'rt') as fp:
...    x = yaml.safe_load(fp)
...
```

SO, I think this should be easy to fix but then we should also think about adding some tests for this :)

- [ ] fix CITATION.cff to be valid YAML; this seems to be as simple as putting `"` double quotes around the title: string.
- [ ] add a test for this, somewhere.

### PR DESCRIPTION
**Pull request by ctb**
Properly quotes the title in `CITATION.cff`, and adds a test to make sure that a YAML parser can parse it.

Fixes https://github.com/sourmash-bio/sourmash/issues/2083

### CODE REVIEW

**Comment by ctb** [comment]
this is failing because (of course) we no longer require yaml. 

I think we can specify that YAML is a test requirement but not an install requirement.

@luizirber thoughts?

**Comment by ctb** [comment]
ok - ready for review @sourmash-bio/devs!

File: tests/test_sourmash.py [hunk 1]
```diff
@@ -33,6 +33,18 @@
...
 
 
+def test_citation_file():
+    import yaml
+
+    thisdir = os.path.dirname(__file__)
+    citation_file = os.path.join(thisdir, '../CITATION.cff')
+
+    with open(citation_file) as fp:
+        x = yaml.safe_load(fp)
```
Comment (hehouts) [inline tests/test_sourmash.py:43]: yaml.safe_load is cool! 
I'm curious what "fp" stands for?

**Review by hehouts — APPROVED** [review]
looks good to me!

File: tests/test_sourmash.py (hunk 1 above)
Comment (ctb) [reply tests/test_sourmash.py:43]: `file pointer`, I think. I think it was used a lot in _Programming Python_, or maybe it's just a holdover from my C programming days.

**Comment by ctb** [comment]
thanks!

**Pull request merged**